### DISCUSSION TITLE
Are custom controls supported?

### DISCUSSION THREAD

**Discussion by k-preslav** [picoe/Eto#2304] [Q&A]
I was wondering if custom winforms controls can be used with eto on any platform? Im trying to build a code editor for my language and need a custom text box that has support for syntax highlight

**Comment by cwensley** [comment]
Yes, custom winforms controls can be used, but only when using Eto's WinForms or even WPF platform (with a little extra plumbing).  Eto does not magically make WinForms code work on all platforms, unless you re-write the WinForms/System.Drawing code to use Eto.Forms/Eto.Drawing instead.
